### PR TITLE
Fixed #15165 -- Prevented wrong results with perimeter on geodetic field

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

**Comment by claudep** [comment]
buildbot, test on oracle

**Comment by timgraham** [comment]
buildbot, test on oracle.

File: tests/gis_tests/distapp/tests.py [hunk 1]
```diff
@@ -685,6 +689,16 @@ def test_perimeter(self):
         for city in qs:
             self.assertEqual(0, city.perim.m)
 
+    @skipUnlessDBFeature("has_Perimeter_function")
+    def test_perimeter_geodetic(self):
+        # Currently no backend support calculating the perimeter on geodetic
```
Comment (timgraham) [inline tests/gis_tests/distapp/tests.py]: no backends support

**Comment by timgraham** [comment]
On Oracle:
```
Traceback (most recent call last):
  File "/mnt/jenkinsdata/workspace/pull-requests-oracle/database/oragis11/python/python3.5/django/test/testcases.py", line 1068, in skip_wrapper
    return test_func(*args, **kwargs)
  File "/mnt/jenkinsdata/workspace/pull-requests-oracle/database/oragis11/python/python3.5/tests/gis_tests/distapp/tests.py", line 697, in test_perimeter_geodetic
    list(CensusZipcode.objects.annotate(perim=Perimeter('poly')))
AssertionError: NotImplementedError not raised
```

**Comment by claudep** [comment]
If you have a local Oracle instance, could you test and report the result of CensusZipcode.objects.annotate(perim=Perimeter('poly'))[0].perim in that test ? I'd like to be sure Oracle behaves like other backends.

**Comment by timgraham** [comment]
Not on my laptop, I'll do it when I return home on the weekend if still needed.

**Comment by claudep** [comment]
@jtiai Maybe you can check that?

**Comment by jtiai** [comment]
@claudep sure, I'll check it when I get back from my vacation.

**Comment by timgraham** [comment]
On Oracle: "18406.3818954314 m"

**Comment by timgraham** [comment]
@jtiai could you advise if we should use the Oracle specific value here?

**Comment by jtiai** [comment]
Yes. It's geographical distance which is due different triangulation algorithms. Relative error between those two values is about 0.011 %

**Comment by timgraham** [comment]
Claude, if you can advise what should be done about Oracle, I can finish it up if needed.

**Comment by claudep** [comment]
The assertion should probably simply made more permissive, as we cannot do anything about the precision difference.

**Comment by timgraham** [comment]
I understand that part, the issue is that Oracle doesn't raise `NotImplementedError` as described in a comment above. Do we need to add a `Perimeter.as_oracle()` method which does that?

**Comment by claudep** [comment]
Sorry. Based on your input, it seems Oracle might support proper calculating a perimeter on geodetic coordinates, right? Then if we don't want to special-case Oracle explicitly in the test, we might add a gis db feature for that.

**Comment by timgraham** [comment]
How does this look? http://dpaste.com/0WDK5X5

**Comment by claudep** [comment]
Looks good. If tests are happy with that, so am I!

**Comment by timgraham** [comment]
Great, I applied my patch in https://github.com/django/django/pull/5901 to verify tests.

**Pull request closed without merging**